extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/Liquibase.java

```diff
@@ -0,0 +1,932 @@
+        return executeFunction(t -> t.calculateCheckSum(changeSetIdentifier));
+    }
+
+    /**
+     * Implementation of the {@link liquibase.Liquibase#calculateCheckSum(String, String, String)} ()} method.
+     * 
+     * @param filename the file name
+     * @param id the change if
+     * @param author the author
+     * @return the check sum of the change
+     * @throws LiquibaseException if the method fails.
+     */
+    public CheckSum calculateCheckSum(String filename, String id, String author) throws LiquibaseException {
+        return executeFunction(t -> t.calculateCheckSum(filename, id, author));
+    }
+
+    /**
+     * Implementation of the {@link liquibase.Liquibase#diff(Database, Database, CompareControl)} ()} method.
+     * 
+     * @param referenceDatabase the reference database
+     * @param targetDatabase the target database
+     * @param compareControl the compare control
+     * @return the diff result
+     * @throws LiquibaseException if the method fails
+     */
+    public DiffResult diff(Database referenceDatabase, Database targetDatabase, CompareControl compareControl)
+            throws LiquibaseException {
+        return executeFunction(t -> t.diff(referenceDatabase, targetDatabase, compareControl));
+    }
+
+    /**
+     * Implementation of the {@link liquibase.Liquibase#validate()} method.
+     * 
+     * @throws LiquibaseException if the method fails.
+     */
+    void validate() throws LiquibaseException {
+        executeConsumer(liquibase.Liquibase::validate);
+    }
+
+    /**
+     * Creates the default labels base on the configuration
+     * 
+     * @return the label expression
+     */
+    private LabelExpression createLabels() {
+        return new LabelExpression(config.labels);
+    }
+
+    /**
+     * Creates the default contexts base on the configuration
+     * 
+     * @return the contexts
+     */
+    private Contexts createContexts() {
+        return new Contexts(config.contexts);
+    }
+
+    /**
+     * Creates the liquibase
+     * 
+     * @param con the database connection
+     * @return the corresponding liquibase instance
+     * @throws DatabaseException if the method fails
+     */
+    private liquibase.Liquibase createLiquibase(Connection con) throws DatabaseException {
+        Database database = DatabaseFactory.getInstance().findCorrectDatabaseImplementation(new JdbcConnection(con));
+        database.setDatabaseChangeLogLockTableName(config.databaseChangeLogLockTableName);
+        database.setDatabaseChangeLogTableName(config.databaseChangeLogTableName);
+        config.currentDateTimeFunction.ifPresent(database::setCurrentDateTimeFunction);
+        config.liquibaseCatalogName.ifPresent(database::setLiquibaseCatalogName);
+        config.liquibaseSchemaName.ifPresent(database::setLiquibaseSchemaName);
+        config.liquibaseTablespaceName.ifPresent(database::setLiquibaseTablespaceName);
+
+        config.defaultCatalogName.ifPresent(d -> {
```

Comment:
Generally, we try to avoid lambdas in runtime code as it has a non negligible memory cost.

Let's get to the bottom of this Liquibase instantiation thing before changing that though.

Reply:
Done